src/lib/fetch.ts

```diff
@@ -4,20 +4,30 @@ import { URL } from 'url';
 import * as fs from 'fs';
 import { readFile } from 'fs/promises';
 
-export async function azleFetch(input: any, init?: any): Promise<any> {
+export async function azleFetch(
+    input: RequestInfo | URL,
+    init?: RequestInit | undefined
+): Promise<Response> {
     if (process.env.AZLE_TEST_FETCH === 'true') {
         console.log('azleFetch has been called');
     }
 
+    if (typeof input !== 'string') {
```

Comment:
I don't think this is necessary. URL objects are pretty easy to turn into strings and it's pretty easy to sort a Request object into a url and most of the options object.

Reply:
http_client/fetch_ic/call.ts has a great function getUrlString that could be moved to a place accessible by both files and then used here. Something like that would be great! Just don't forget to also look for the options in that Request object

Reply:
please add URL and add a comment about not adding Request

Reply:
This restriction is gone, I made a new function called `getUrl` and added the comment about Request there